src/jansson.h

```diff
@@ -126,6 +126,9 @@ typedef struct {
     char text[JSON_ERROR_TEXT_LENGTH];
 } json_error_t;
 
+char *json_error_get_detailed(json_error_t *error, const char *src, int color);
+char *json_error_get_arrow(json_error_t *error, const char *src, int length, int color);
+char *json_error_get_source_text(json_error_t *error, const char *src);
```

Comment:
I'm not sure whether all of these need to be a part of the public API. May we should start by exposing `json_error_get_detailed` and a function that prints directly to the given stream (stdout, stderr, or user-supplied FILE pointer).

Minimizing the public API makes maintaining easier. With more public functions you get more bugs reported :) Other functions can be added to the public API later if needed.